cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultSchedulerService.java

```diff
@@ -68,7 +57,7 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
       String key = trigger.getKey().getName();
       String[] parts = key.split(":");
       Preconditions.checkArgument(parts.length == 6, String.format("Trigger's key name %s has %d parts instead of 6",
```

Comment:
the separator here is ":" but ScheduleId.fromIdParts() parses with separator "." I'll add a helper method in AbstractSchedulerService